### PR TITLE
[shared_preferences_platform_interface] Migrate to null safety

### PR DESCRIPTION
**Pull request by yash1200**
## Description

Migrates `shared_preferences_platform_interface` to null safety.

## Related Issues

Fixes flutter/flutter#70219

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by cyanglaz** [comment]
CC @blasten

File: packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart [hunk 1]
```diff
@@ -47,7 +47,7 @@ abstract class SharedPreferencesStorePlatform {
   bool get isMock => false;
 
   /// Removes the value associated with the [key].
-  Future<bool> remove(String key);
+  Future<bool?> remove(String key);
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart:50]: When does `remove` return `null`, and what does it mean in terms of a consumer of this API?

File: packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart [hunk 2]
```diff
@@ -58,13 +58,13 @@ abstract class SharedPreferencesStorePlatform {
   /// * Value type "Int" must be passed if the value is of type `int`.
   /// * Value type "String" must be passed if the value is of type `String`.
   /// * Value type "StringList" must be passed if the value is of type `List<String>`.
-  Future<bool> setValue(String valueType, String key, Object value);
+  Future<bool?> setValue(String valueType, String key, Object value);
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart:61]: ditto

File: packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart [hunk 3]
```diff
@@ -58,13 +58,13 @@ abstract class SharedPreferencesStorePlatform {
...
 
   /// Removes all keys and values in the store.
-  Future<bool> clear();
+  Future<bool?> clear();
 
   /// Returns all key/value pairs persisted in this store.
-  Future<Map<String, Object>> getAll();
+  Future<Map<String, Object>?> getAll();
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart:67]: ditto

File: packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart [hunk 4]
```diff
@@ -58,13 +58,13 @@ abstract class SharedPreferencesStorePlatform {
   /// * Value type "Int" must be passed if the value is of type `int`.
   /// * Value type "String" must be passed if the value is of type `String`.
   /// * Value type "StringList" must be passed if the value is of type `List<String>`.
-  Future<bool> setValue(String valueType, String key, Object value);
+  Future<bool?> setValue(String valueType, String key, Object value);
 
   /// Removes all keys and values in the store.
-  Future<bool> clear();
+  Future<bool?> clear();
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart:64]: ditto

File: packages/shared_preferences/shared_preferences_platform_interface/test/method_channel_shared_preferences_test.dart [hunk 5]
```diff
@@ -15,41 +15,41 @@ void main() {
...
+    const Map<String, Object> kTestValues = <String, Object>{
       'flutter.String': 'hello world',
       'flutter.Bool': true,
       'flutter.Int': 42,
       'flutter.Double': 3.14159,
       'flutter.StringList': <String>['foo', 'bar'],
     };
 
-    InMemorySharedPreferencesStore testData;
+    InMemorySharedPreferencesStore? testData;
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/test/method_channel_shared_preferences_test.dart:26]: use `late` here

File: packages/shared_preferences/shared_preferences_platform_interface/test/method_channel_shared_preferences_test.dart [hunk 6]
```diff
@@ -15,41 +15,41 @@ void main() {
...
 
     final List<MethodCall> log = <MethodCall>[];
-    MethodChannelSharedPreferencesStore store;
+    MethodChannelSharedPreferencesStore? store;
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_platform_interface/test/method_channel_shared_preferences_test.dart:29]: `late`

**Comment by blasten** [comment]
Thanks for the PR! We have merged the `nnbd` branch into master, so I suggest reopening a PR against master after addressing the comments.

**Comment by yash1200** [comment]
Sure I'll reopen the PR on master branch.

**Pull request closed without merging**